### PR TITLE
docs: add web build info

### LINKED ISSUE #829: [DOC] Add GitHub page: How the docs are built
### Change in the documentation

To minimise the learning curve for potential new docs contributors, it's important to explain exactly how, where, when and why the https://flagd.dev comes into being.

Starting this exercise early (while the website is new and relatively simple) will make it easier in the long term.

TODO:

- Add a page to GiHub explaining the mechanics of how the documentation website is built
- Updating this page should be included in the "definition of done" for any update in the **way** the website is built (ie. new PR checks are added or removed, docs tooling is changed etc.)

### PR DESCRIPTION
**Pull request by agardnerIT**
## This PR
- Adds a new page explaining how the website is built
- Provides info to would-be contributors on how to get started
- Closes #829

### Notes

This work has identified some improvements that can be made in the GitHub Actions setup.

See #835 for more info.

### Follow-up Tasks

@thomaspoignant and @beeme1mr to assist on the TODO items (see comments in the markdown file)

### CODE REVIEW

File: docs/WEB_BUILD_INFO.md
Comment (beeme1mr) [inline docs/WEB_BUILD_INFO.md:1]: Please move this to the contributing doc.

File: docs/WEB_BUILD_INFO.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+This guide is intended so that potential contributors can understand how, where and when the website comes into being.
+
+## Website Structure
+
+The website is built on [mkdocs](https://www.mkdocs.org/) and the [material theme](https://squidfunk.github.io/mkdocs-material/).
+
+The [mkdocs.yml](../mkdocs.yml) file describes the high level metadata in addition to the main menu structure.
+
+The website is generated from files in the [web-docs](../web-docs/) folder.
```
Comment (beeme1mr) [inline docs/WEB_BUILD_INFO.md:13]:
```suggestion
The website is generated from files in the [docs](../docs/) folder.
```

File: docs/WEB_BUILD_INFO.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+The website homepage is powered by the [index.md](../web-docs/index.md) file at the root of `/web-docs`
```
Comment (beeme1mr) [inline docs/WEB_BUILD_INFO.md:15]:
```suggestion
```

File: docs/WEB_BUILD_INFO.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+## What happens during a Website-Related PR?
+
+Imagine you notice a typo on [https://flagd.dev](https://flagd.dev).
+
+You find the page inside [/web-docs](../web-docs/) and fix the typo and commit the change (remembering to sign your commit).
```
Comment (beeme1mr) [inline docs/WEB_BUILD_INFO.md:21]:
```suggestion
Find the page inside [/docs](../docs/) and fix the typo then commit the change (remembering to sign your commit).
```

Clicking the pencil icon on the doc page may be the easy way to find the correct page.

File: docs/WEB_BUILD_INFO.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+   Click on the Details hyperlink in the PR and scroll to the bottom of the resulting page.
+1. PR titles must be semantically correct.
+   [The lint-pr GitHub action](../.github/workflows/lint-pr.yaml) checks that and warns if your PR title doesn't match requirements.
+1. The markdown files are linted in [the markdown-checks GitHub Action](../.github/workflows/markdown-checks.yaml) which calls [make markdownlint](../Makefile#L103).
+   Your markdown syntax [must follow a set of rules](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md).
+   Hint: Run `make markdownlint-fix` and / or `make markdownlint` before raising the PR!
+1. [The build GitHub Action](../.github/workflows/build.yaml) runs a job called [lint](../.github/workflows/build.yaml#L23)
+   which in turn runs [make workspace-init](../Makefile#L12) and then [make lint](../Makefile#L62).
+   TODO: @thomaspoignant to advise on what this does and if it's still relevant for documentation.
+   That linting _appears_ to me to be related to Go code, not docs.
```
Comment (beeme1mr) [inline docs/WEB_BUILD_INFO.md:39]: The linking issues have been addressed.

File: docs/WEB_BUILD_INFO.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+   All of which appear irrelevant to the documentation.
+
+   TODO: Rebuild / rescope to remove from documentation PRs?
+1. A preview version of the site is built by Netflify.
+1. The Netlify preview lint is added to the PR.
+1. Netlify will rebuild the site for every new commit.
+
+## How does the site get built and where is it deployed?
+
+@beeme1mr to help author this section.
```
Comment (beeme1mr) [inline docs/WEB_BUILD_INFO.md:63]: Netlify has been configured to look for the [netlify.toml](https://github.com/open-feature/flagd/blob/main/netlify.toml) file at the root of the project and deploy according to that configuration.

File: docs/WEB_BUILD_INFO.md
Comment (toddbaert) [reply docs/WEB_BUILD_INFO.md:1]: Ya, agreed I think that's a better place.

**Comment by beeme1mr** [comment]
Closing due to inactivity.

**Pull request closed without merging**